test/controllers/beneficial.owner.delete.warning.controller.spec.ts

```diff
@@ -198,4 +244,124 @@ describe("BENEFICIAL OWNER DELETE WARNING controller", () => {
       expect(resp.text).toContain(SERVICE_UNAVAILABLE);
     });
   });
+
+  describe("POST with url params tests", () => {
+    test("redirects to the beneficial owner type page if No option has been selected", async () => {
+      const resp = await request(app)
+        .post(config.BENEFICIAL_OWNER_DELETE_WARNING_WITH_PARAMS_URL)
+        .send({ delete_beneficial_owners: "0" });
+
+      expect(resp.status).toEqual(302);
+      expect(mockSetExtraData).not.toHaveBeenCalled();
+      expect(mockCheckBOsDetailsEntered).not.toHaveBeenCalled();
+      expect(mockCheckMOsDetailsEntered).not.toHaveBeenCalled();
+    });
+
+    test(`redirects to the beneficial owner type page if Yes option has been selected and
+        ${BeneficialOwnersStatementType.SOME_IDENTIFIED_ALL_DETAILS} as statement type`, async () => {
+
+      mockIsActiveFeature.mockReturnValueOnce(true); // For FEATURE_FLAG_ENABLE_REDIS_REMOVAL
+      mockGetApplicationData.mockReturnValueOnce(APPLICATION_DATA_MOCK);
+
+      const resp = await request(app)
+        .post(config.BENEFICIAL_OWNER_DELETE_WARNING_WITH_PARAMS_URL)
+        .send({
+          delete_beneficial_owners: "1",
+          BeneficialOwnerStatementKey: BeneficialOwnersStatementType.SOME_IDENTIFIED_ALL_DETAILS
+        });
+
+      expect(resp.status).toEqual(302);
+      expect(resp.text).toContain(NEXT_PAGE_URL);
+      expect(resp.header.location).toEqual(NEXT_PAGE_URL);
+      expect(mockSetExtraData).toHaveBeenCalledTimes(1);
+      expect(mockGetUrlWithParamsToPath.mock.calls[0][0]).toEqual(config.BENEFICIAL_OWNER_TYPE_WITH_PARAMS_URL);
```

Comment:
I think you should assert this too:

`expect(mockGetUrlWithParamsToPath).toHaveBeenCalledTimes(1);`

As we have done elsewhere.

Reply:
Added this assertion.